src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.h

```diff
@@ -0,0 +1,346 @@
+     * iteration stops and returns that same error result.
+     */
+    virtual CHIP_ERROR RootCertsForFabric(EndpointId matterEndpoint, FabricIndex fabric,
+                                          RootCertificateListCallback loadedCallback) const = 0;
+
+    /**
+     * @brief Finds the TLSCertStruct with the given id. The certificates passed to loadedCallback has a
+     * guaranteed lifetime of the method call.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] id The id of the root certificate to find.
+     * @param[out] loadedCallback The lambda to execute with the loaded root cert.
+     * @return CHIP_ERROR_NOT_FOUND if no mapping is found.
+     */
+    virtual CHIP_ERROR FindRootCert(EndpointId matterEndpoint, FabricIndex fabric, Tls::TLSCAID id,
+                                    LoadedRootCertificateCallback loadedCallback) const = 0;
+
+    /**
+     * @brief Finds the TLSCertStruct with the given fingerprint. The certificates passed to loadedCallback has a
+     * guaranteed lifetime of the method call.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] fingerprint The fingerprint of the root certificate to find.
+     * @param[out] loadedCallback The lambda to execute with the loaded root cert.
+     * @return CHIP_ERROR_NOT_FOUND if no mapping is found.
+     */
+    virtual CHIP_ERROR LookupRootCertByFingerprint(EndpointId matterEndpoint, FabricIndex fabric, const ByteSpan & fingerprint,
+                                                   LoadedRootCertificateCallback loadedCallback) const = 0;
+
+    /**
+     * @brief Like LookupRootCertByFingerprint except looks up by the certificate instead of fingerprint.
+     * Implementations will generally fingerprint the certificate & delegate.
+     */
+    virtual CHIP_ERROR LookupRootCert(EndpointId matterEndpoint, FabricIndex fabric, const ByteSpan & cert,
+                                      LoadedRootCertificateCallback loadedCallback) const = 0;
+    /**
+     * @brief Removes the root certificate with the given ID
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] id The id of the root certificate to remove.
+     * @return CHIP_ERROR_NOT_FOUND if no mapping is found.
+     */
+    virtual CHIP_ERROR RemoveRootCert(EndpointId matterEndpoint, FabricIndex fabric, Tls::TLSCAID id) = 0;
+
+    /**
+     * @brief Generates a client certificate signing request (CSR)
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fingerprint The fingerprint of the root certificate to find.
+     * @param[out] loadedCallback The lambda to execute with the generated client CSR.
```

Comment:
Maybe I should have been clearer in https://github.com/project-chip/connectedhomeip/pull/39754#discussion_r2176140536 : The docs need to say that the loadedCallback needs to be called before this function returns.

Also, it's not an `[out]` param....